polus-feature-extraction-plugin/src/main.py

```diff
@@ -1097,130 +1075,218 @@ def main():
+            
+    #Get list of .ome.tif files in the directory including sub folders for labeled images
+    # Try to infer a filepattern from the files on disk for faster matching later
+    if segDir:
+        configfiles_seg = filepattern.FilePattern(segDir,pattern)
+        files_seg = list(configfiles_seg())
+    else:
+        label_image = None
+
+    #Get list of .ome.tif files in the directory including sub folders for intensity images
+    if intDir:      
+        configfiles_int = filepattern.FilePattern(intDir,pattern)
+        files_int = list(configfiles_int())
+    else:
+        intensity_image = None
         
-        #Get list of .ome.tif files in the directory including sub folders for intensity images
-        if intDir:
-            configfiles_int = list_file(intDir)
-            #Check whether .ome.tif files are present in the intensity image directory
-            if not configfiles_int:
-                raise ValueError('No intensity image .ome.tif files found.')
-        else:
-            intensity_image = None
-   
+    #Check for matching filepattern
+    if segDir and intDir:
+        if len(files_seg) == 0 and len(files_int)==0 :
+            raise ValueError("Could not find files matching filepattern")
+    elif segDir and not intDir:
+        if len(files_seg) == 0:
+            raise ValueError("Could not find labeled/segmented image files matching filepattern")
+    elif intDir and not segDir:
+        if len(files_int) == 0:
+            raise ValueError("Could not find intensity image files matching filepattern")
+
         
+    if not segDir:
+        for intfile in configfiles_int():
+            df=None
+            logger.info('Started reading the file: {}'.format(intfile[0]['file'].name))
+            with BioReader(intfile[0]['file']) as br:
+                intensity_image = br[:,:,0:1,0,0].squeeze()
+            logger.info('Done reading the file: {}'.format(intfile[0]['file'].name))
+            df,title = feature_extraction(features,
+                                                intfile[0]['file'],
+                                                embeddedpixelsize,
+                                                unitLength,
+                                                pixelsPerunit,
+                                                pixelDistance,
+                                                1,
+                                                intensity_image,
+                                                label_image=None,
+                                                img_emb_uint=None,
+                                                )
+            os.chdir(outDir)
+            if csvfile == 'separatecsv':
+                logger.info('Saving dataframe to csv for' + intfile[0]['file'].name)
+                export_csv = df.to_csv(r'%s.csv'%title, index=None, header=True, encoding='utf-8-sig')
+            else:
+                df_csv = df_csv.append(df)
```

Comment:
It seems like there is a lot of code replication going on here and below. You have multiple if statements that have large blocks of code that is mostly similar between them. You should consider trying to minimize the amount of code running inside of these if blocks so that future modifications to the code are easier. For example, maybe only have one call to `feature_extraction` and just modify the inputs you provide that inside of different `if` blocks.